tools/expotools/src/Changelogs.ts

```diff
@@ -157,131 +171,137 @@ export class Changelog {
         continue;
       }
 
-      if (currentVersion && currentSection && token.type === Markdown.TokenType.LIST_ITEM_START) {
-        i++;
-        for (; tokens[i].type !== Markdown.TokenType.LIST_ITEM_END; i++) {
-          const token = tokens[i] as Markdown.TextToken;
+      if (currentVersion && currentSection && token.type === Markdown.TokenType.LIST) {
+        for (const item of token.items) {
+          const text =
+            item.tokens.find(
+              (token): token is Markdown.TextToken => token.type === Markdown.TokenType.TEXT
+            )?.text ?? item.text;
 
-          if (token.text) {
-            changes.totalCount++;
-            versions[currentVersion][currentSection].push(token.text);
-          }
+          changes.totalCount++;
+          versions[currentVersion][currentSection].push(text.trim());
         }
       }
     }
     return changes;
   }
 
   /**
-   * Inserts given entry to changelog.
+   * Saves changes that we made in the array of tokens.
    */
-  async addChangeAsync(entry: ChangelogEntry): Promise<void> {
-    const tokens = [...(await this.getTokensAsync())];
-    const sectionIndex = tokens.findIndex(
-      (token) =>
-        token.type === Markdown.TokenType.HEADING &&
-        token.depth === VERSION_HEADING_DEPTH &&
-        token.text === entry.version
-    );
+  async saveAsync(): Promise<void> {
+    // If tokens where not loaded yet, there is nothing to save.
+    if (!this.tokens) {
+      return;
+    }
```

Comment:
Is it called in this incorrect state anywhere? Maybe it would be better to throw upon encountering incorrect state?

Reply:
That's fine as is I think. Tokens are lazy-loaded (it's `null` before loading) and if they weren't loaded yet then there is also nothing to save.
Imho sometimes it's better not to throw errors in some specific cases which we don't expect to happen but once they happen the expected **state** after calling this method is the same.